extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeCommentsExtractor.java

```diff
@@ -41,53 +43,99 @@
     private String ytClientName;
     private String responseBody;
 
-    public YoutubeCommentsExtractor(StreamingService service, ListLinkHandler uiHandler) {
+    private Optional<Boolean> optCommentsDisabled = Optional.empty();
```

Comment:
Why not just set this to `false` by default instead of using an `Optional` and then `orElse(false)`?

Reply:
@Stypox 

TL;DR

> Caching mechanism and holder of the commentsDisabled value.
> 
> Initial value = empty -> unknown if comments are disabled or not
> Some method calls {@link YoutubeCommentsExtractor#findInitialCommentsToken()} → value is set.
> If the method or another one that is depending on disabled comments is now called again, the method execution can avoid unnecessary calls.

Example:
* A new ``YoutubeCommentsExtractor`` is initialized
* ``isCommentsDisabled`` is called -> initialization in ``findInitialCommentsToken`` occurs 
* As it turns out the comments are disabled and therefore ``optCommentsDisabled`` is set to ``true``
* Shortly after that``getInitialPage`` is called -> we can instantly return the ``getInfoItemsPageForDisabledComments`` without processing ``findInitialCommentsToken`` again

Another example (which matches your question):
* A new ``YoutubeCommentsExtractor`` is initialized
* ``getInitialPage`` is called -> initialization in ``findInitialCommentsToken`` occurs 
* As it turns out the comments are NOT disabled and therefore ``optCommentsDisabled`` is set to ``false``
* Shortly after that ``isCommentsDisabled`` is called ->  we can instantly return the value of ``optCommentsDisabled`` because we already know it 
* if ``optCommentsDisabled`` would be initialized by default with ``false``, it's unknown if ``findInitialCommentsToken`` was already called and resulted in ``false`` or not and therefore the ``findInitialCommentsToken`` should then be executed again

Reply:
Ok, I see, this part looks good then